api/providers/urls.py

```diff
@@ -0,0 +1,16 @@
+from django.conf.urls import url
+
+from api.providers import views
+
+app_name = 'osf'
+
+urlpatterns = [
+    url(r'^preprints/$', views.PreprintProviderList.as_view(), name=views.PreprintProviderList.view_name),
+    url(r'^preprints/(?P<provider_id>\w+)/$', views.PreprintProviderDetail.as_view(), name=views.PreprintProviderDetail.view_name),
+    url(r'^preprints/(?P<provider_id>\w+)/licenses/$', views.PreprintProviderLicenseList.as_view(), name=views.PreprintProviderLicenseList.view_name),
+    url(r'^preprints/(?P<provider_id>\w+)/preprints/$', views.PreprintProviderPreprintList.as_view(), name=views.PreprintProviderPreprintList.view_name),
```

Comment:
I have a feeling that `provider/preprints/osf/preprints` is a bit weird now. Should <preprints> be a generic term for things that are collected by this provider? @mfraezz can decide that.

Reply:
Agree that it should be a generic term. `provided`, `collected`, either of those with `_objects` appended (or just `objects`), `submissions`? ¯\\\_(ツ)_/¯ 

I don't have strong feelings, @brianjgeiger might.

Reply:
Haven't you heard, naming things is easy. 😆 
I think I hate `submissions` the least.

Maybe we should just call it `service` a la `PreprintService` just to keep the dream alive.

Reply:
Hmmmmmm. I lean towards just going ahead and leaving it, because one day you may want a provider to be able to provide more than one kind of thing (and not just in the `linked_guids` kind of way). It's also clear what kind of thing you're going to get back, which will likely make it easier for routing in ember (i.e. you're using the preprints model on things coming back from this route), though that's likely to be handled by the relationship name.

Reply:
It would be nice to be able to write generic views for different provider types, given that they would all behave approximately the same.

Reply:
I think that's a great idea. Most likely, while the serializers may be somewhat different, the view probably won't be. There's no reason you couldn't write one view and use it for each of the URLs.